src/java_bytecode/java_types.h

```diff
@@ -609,12 +609,18 @@ class java_generic_symbol_typet : public symbol_typet
   /// in the vector of generic types.
   /// \param type The type we are looking for.
   /// \return The index of the type in the vector of generic types.
-  optionalt<size_t> generic_type_index(const reference_typet &type) const
+  optionalt<size_t>
+  generic_type_index(const java_generic_parametert &type) const
   {
-    const auto &type_pointer =
-      std::find(generic_types().begin(), generic_types().end(), type);
-    if(type_pointer != generic_types().end())
-      return type_pointer - generic_types().begin();
+    const auto &type_variable = type.type_variable();
```

Comment:
Instead of `type_variable` use `get_name()` to get the unique name of the parameter.